docs/lesson-plans/week-4.md

```diff
@@ -0,0 +1,612 @@
+client.
+
+### Postman and API Clients
+
+Now that you know some of the basics of web requests and the way that the web works, let's try out some HTTP requests
+and inspect the responses. API clients are programs that nicely let you send any kind of request with any information to
+whatever URLs you define. Some of the nice ones even let you define setup scripts or post request tests for ease of use
+and validation later on.
+
+In terms of being able to be run through the command line and being on almost every computer, 
+[curl](https://curl.haxx.se/) is an extremely useful and functional (but quite complicated) tool. Feel free to try it 
+out with a command like `curl https://curl.haxx.se/`, and you should see the curl webpage in HTML. If you try 
+`curl -o test.html https://curl.haxx.se/`, and then open that file in a web browser, you should see the webpage, but
+downloaded!
+
+Working entirely on the command line like that can get really annoying though, so try out one of our favorite API 
+clients: [Postman](https://www.postman.com/). In here, you can run almost any type of request, authenticate, set
+headers and body (if applicable), run scripts and tests, see the output (in HTML too!), status code, time, size, and
+returned cookies and headers! If you find that you end up using a lot of the same requests, you can also save them and 
+pull them up later on. We've used this for classes, coop, and personal projects, and hopefully you'll find this useful.
+
+## Routing on the Backend
+
+Now we'll get to setting up the individual routes for the backend of our website so that you can access different 
+resources.
+
+### Structure of the Backend
+
+From the root directory, you'll notice that there are three main directories: api, persist, and service. 
+
+The api directory is where all requests will enter through. It is the way that any external program will interact with
+and view the results of our application.
+
+The persist directory is where all *persisting* information will live. Starting out, it is where we'll create our 
+database mocks, and later on, it will be where our database interactivity classes will be placed.
+
+The service directory is what unifies everything together. Since we want to not make the persist and api directories 
+dependent on each other, there will end up being a few interfaces that are made available and implemented in service. 
+This is because some of the operations we want to perform need to know of both persist and api functionality (but 
+remember, they shouldn't know about each other). 
+
+Don't worry about having to figure all of this out yourself though, we'll provide some help to get you started. Also,
+for those of you who know about the 
+[Model-View-Controller](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) design pattern, this 
+follows that! In this case, persist is our model (since it stores and holds the data), api is our view (since it's what
```

Comment:
add a short description of what MVC is for those who don't know; it may get a little discouraging?

or mention if you don't know, it's completely fine, we'll cover that soon

Reply:
I added a quick explanation below.